pennylane/_qutrit_device.py

```diff
@@ -0,0 +1,390 @@
+
+        # reshape the probability so that each axis corresponds to a wire
+        prob = self._reshape(prob, [3] * self.num_wires)
+
+        # sum over all inactive wires
+        prob = self._flatten(self._reduce_sum(prob, inactive_device_wires.labels))
+        # TODO: Add case for when inactive_device_wires is not an instance of Wires once
+        # default qutrit device is added
+
+        # The wires provided might not be in consecutive order (i.e., wires might be [2, 0]).
+        # If this is the case, we must permute the marginalized probability so that
+        # it corresponds to the orders of the wires passed.
+        num_wires = len(device_wires)
+        basis_states = self.generate_basis_states(num_wires)
+        basis_states = basis_states[:, np.argsort(np.argsort(device_wires))]
+
+        powers_of_three = 3 ** np.arange(len(device_wires))[::-1]
+        perm = basis_states @ powers_of_three
+        return self._gather(prob, perm)
+
+    def sample(self, observable, shot_range=None, bin_size=None):
+
+        # TODO: Add special cases for any observables that require them once list of
+        # observables is updated.
+
+        # translate to wire labels used by device
+        device_wires = self.map_wires(observable.wires)
+        name = observable.name  # pylint: disable=unused-variable
+        sample_slice = Ellipsis if shot_range is None else slice(*shot_range)
+
+        if isinstance(
+            observable, MeasurementProcess
+        ):  # if no observable was provided then return the raw samples
+            if (
+                len(observable.wires) != 0
+            ):  # if wires are provided, then we only return samples from those wires
+                samples = self._samples[sample_slice, np.array(device_wires)]
+            else:
+                samples = self._samples[sample_slice]
+
+        else:
+
+            # Replace the basis state in the computational basis with the correct eigenvalue.
+            # Extract only the columns of the basis samples required based on ``wires``.
+            samples = self._samples[
+                sample_slice, np.array(device_wires)
+            ]  # Add np.array here for Jax support.
+            powers_of_three = 3 ** np.arange(samples.shape[-1])[::-1]
+            indices = samples @ powers_of_three
+            indices = np.array(indices)  # Add np.array here for Jax support.
+            try:
+                samples = observable.eigvals()[indices]
+            except qml.operation.EigvalsUndefinedError as e:
+                # if observable has no info on eigenvalues, we cannot return this measurement
+                raise qml.operation.EigvalsUndefinedError(
+                    f"Cannot compute samples of {observable.name}."
+                ) from e
+
+        if bin_size is None:
+            return samples
+
+        return samples.reshape((bin_size, -1))
+
+    # TODO: Implement function
+    def adjoint_jacobian(
```

Comment:
It looks like this one also only needs a few 2->3 substitutions, but will need to wait until more operations are implemented.

Reply:
Yes, I agree.